climbing_stairs/climbing_stairs.py

```diff
@@ -2,13 +2,28 @@
 
 import sys
 
+
 def climbing_stairs(n, cache=None):
-  pass 
+    if n == 0 or n == 1:
+        return 1
+
+    if n == 2:
+        return 2
+
+    if n == 3:
+        return 4
+
+    return climbing_stairs(n - 1) + climbing_stairs(n - 2) + climbing_stairs(n - 3)
```

Comment:
This is close, you'll also want to implement a cache that returns the cache entry if found, or that creates it otherwise as a result of three recursive calls to the function of `n-1`, `n-2`, and `n-3`. Beej went over this yesterday.